<!DOCTYPE html PUBLIC "-//W3C//DTD Html 1.0 Strict//EN" "http://www.w3.org/TR/Html1/DTD/strict.dtd">
<html><head>
<title>Pigs</title>

<link crossorigin="anonymous" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css" integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm" rel="stylesheet"/>
<script crossorigin="anonymous" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" src="https://code.jquery.com/jquery-3.3.1.slim.min.js"></script>
<script crossorigin="anonymous" integrity="sha384-UO2eT0CpHqdSJQ6hJty5KVphtPhzWj9WO1clHTMGa3JDZwrnQq4sF86dIHNDz0W1" src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.7/umd/popper.min.js"></script>
<script crossorigin="anonymous" integrity="sha384-JjSmVgyd0p3pXB1rRibZUAYoIIy6OrQ6VrjIEaFf/nJGzIxFDsf4x0xIM+B07jRM" src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js"></script>
<link href="/css/bootstrap.css" rel="stylesheet"/>
<link href="images/favicon.ico" rel="icon" type="image/x-icon"/>
<meta content="width=device-width, initial-scale=1" name="viewport"/>
</head>
<body>
<nav class="navbar navbar-expand sticky-top navbar-light bg-light">
<a class="navbar-brand" href="/"><img src="/images/logo.svg"/>MultiBUGS</a>
<div class="collapse navbar" id="navbarSupportedContent">
<ul class="navbar-nav mr-auto">
<li class="nav-item">
<a class="nav-link" href="/">Home</a>
</li>
<li class="nav-item">
<a class="nav-link" href="/about">About</a>
</li>
<li class="nav-item">
<a class="nav-link" href="/download">Download</a>
</li>
<li class="nav-item">
<a class="nav-link" href="/documentation/latest/">Documentation</a>
</li>
<li class="nav-item active">
<a class="nav-link" href="/examples/latest/">Examples <span class="sr-only">(current)</span></a>
</li>
</ul>
</div>
</nav><div class="container-fluid"><div class="row main"><div class="col-md-3 col-lg-2 col-xl-3"><div class="sticky-top stickymenu" id="contents">
<button aria-controls="navbarContents" aria-expanded="false" aria-label="Toggle docs navigation" class="btn btn-link d-md-none ml-3 mr-3 mt-3 collapsed mb-toggler" data-target="#navbarContents" data-toggle="collapse" type="button">
<span class="mb-toggler-icon mr-1"></span>
    Contents
  </button>

<h5 class="d-none d-md-block">Examples</h5>
<div class="collapse mb-links" id="navbarContents">
<ul class="list-unstyled">
<li class="nav-item"><a aria-expanded="true" class="dropdown-toggle nav-link" data-toggle="collapse" href="#volume1">Volume I</a>
<ul class="collapse" id="volume1">
<li><a href="/examples/latest/Volumei.html">Contents</a></li>
<li><a href="/examples/latest/Rats.html" title="Normal hierarchical model">Rats</a></li>
<li><a href="/examples/latest/Pumps.html" title="Conjugate gamma-Poisson hierarchical model">Pumps</a></li>
<li><a href="/examples/latest/Dogs.html" title="Log linear binary model">Dogs</a></li>
<li><a href="/examples/latest/Seeds.html" title="Random effects logistic regression">Seeds</a></li>
<li><a href="/examples/latest/Surgical.html" title="Institutional ranking">Surgical</a></li>
<li><a href="/examples/latest/Magnesium.html" title="Meta-analysis prior sensitivity">Magnesium</a></li>
<li><a href="/examples/latest/Salm.html" title="Extra-Poisson variation in dose-response study">Salm</a></li>
<li><a href="/examples/latest/Equiv.html" title="Bioequivalence in a cross-over trial">Equiv</a></li>
<li><a href="/examples/latest/Dyes.html" title="Variance components model">Dyes</a></li>
<li><a href="/examples/latest/Stacks.html" title="Robust and ridge regression">Stacks</a></li>
<li><a href="/examples/latest/Epil.html" title="Repeated measures on Poisson counts">Epil</a></li>
<li><a href="/examples/latest/Blockers.html" title="Random effects meta-analysis of clinical trials">Blockers</a></li>
<li><a href="/examples/latest/Oxford.html" title="Smooth fit to log-odds ratios in case control studies">Oxford</a></li>
<li><a href="/examples/latest/Lsat.html" title="Latent variable models for item-response data">LSAT</a></li>
<li><a href="/examples/latest/Bones.html" title="Latent trait model for multiple ordered categorical responses">Bones</a></li>
<li><a href="/examples/latest/Inhalers.html" title="Random effects model for ordinal responses from a cross-over trial">Inhalers</a></li>
<li><a href="/examples/latest/Mice.html" title="Weibull regression in censored survival analysis">Mice</a></li>
<li><a href="/examples/latest/Kidney.html" title="Weibull regression with random effects">Kidney</a></li>
<li><a href="/examples/latest/Leuk.html" title="Survival analysis using Cox regression">Leuk</a></li>
<li><a href="/examples/latest/Leukfr.html" title="Cox regression with frailties">LeukFr</a></li>
</ul>
<li class="nav-item"><a aria-expanded="true" class="dropdown-toggle nav-link" data-toggle="collapse" href="#volume2">Volume II</a>
<ul class="collapse show" id="volume2">
<li><a href="/examples/latest/Volumeii.html">Contents</a></li>
<li><a href="/examples/latest/Dugongs.html" title="A nonconjugate, nonlinear model">Dugongs</a></li>
<li><a href="/examples/latest/Otrees.html" title="A hierarchical, nonlinear model">Orange trees</a></li>
<li><a href="/examples/latest/OtreesMVN.html" title="A hierarchical, nonlinear model">MV Orange trees</a></li>
<li><a href="/examples/latest/Biopsies.html" title="Latent class model">Biopsies</a></li>
<li><a href="/examples/latest/Eyes.html" title="Normal mixture model">Eyes</a></li>
<li><a href="/examples/latest/Hearts.html" title="A mixture model for count data">Hearts</a></li>
<li><a href="/examples/latest/Air.html" title="Covariate measurement error">Air</a></li>
<li><a href="/examples/latest/Cervix.html" title="Case-control study with errors in covariates">Cervix</a></li>
<li><a href="/examples/latest/Jaws.html" title="Repeated measures analysis of variance">Jaw</a></li>
<li><a href="/examples/latest/BiRats.html" title="A bivariate Normal hierarchical model">Birats</a></li>
<li><a href="/examples/latest/Schools.html" title="Multivariate hierarchical model of examination results">Schools</a></li>
<li><a href="/examples/latest/Ice.html" title="Non-parametric smoothing in an age-cohort model">Ice</a></li>
<li><a href="/examples/latest/Beetles.html" title="Logistic, probit and extreme value models">Beetles</a></li>
<li><a href="/examples/latest/Alligators.html" title="Multinomial logistic model">Alligators</a></li>
<li><a href="/examples/latest/Endo.html" title="Conditional inference in case-control studies">Endo</a></li>
<li><a href="/examples/latest/Stagnant.html" title="A change point problem">Stagnant</a></li>
<li><a href="/examples/latest/Asia.html" title="An expert system">Asia</a></li>
<li>Pigs</li>
<li><a href="/examples/latest/t-df.html" title="Degree of freedom of a t distribution">Simulating data</a></li>
</ul>
<li class="nav-item"><a aria-expanded="true" class="dropdown-toggle nav-link" data-toggle="collapse" href="#volume3">Volume III</a>
<ul class="collapse" id="volume3">
<li><a href="/examples/latest/Volumeiii.html">Contents</a></li>
<li><a href="/examples/latest/Camel.html" title="Multivariate normal with structured missing data">Camel</a></li>
<li><a href="/examples/latest/Eyetracking.html" title="Dirichlet process prior">Eye Tracking</a></li>
<li><a href="/examples/latest/Fire.html" title="Data distribution using dloglik">Fire insurance claims</a></li>
<li><a href="/examples/latest/Funshapes.html" title="General constraints">Fun Shapes</a></li>
<li><a href="/examples/latest/Hepatitis.html" title="Random effects model with measurement error">Hepatitis</a></li>
<li><a href="/examples/latest/Hips1.html" title="Deterministic cost-effectiveness model">Hips1</a></li>
<li><a href="/examples/latest/Hips2.html" title="Stochastic cost-effectiveness model">Hips2</a></li>
<li><a href="/examples/latest/Hips3.html" title="Cost-effectiveness model">Hips3</a></li>
<li><a href="/examples/latest/Hips4.html" title="Cost-effectiveness model">Hips4</a></li>
<li><a href="/examples/latest/Jama.html" title="Radio carbon dating with phase information">Jama</a></li>
<li><a href="/examples/latest/Pigweights.html" title="Smoothed histogram">Pig Weights</a></li>
<li><a href="/examples/latest/Pines.html" title="Bayes factors using pseudo priors">Pines</a></li>
<li><a href="/examples/latest/StVeit.html" title="Radio carbon dating with stratification">St Veit</a></li>
</ul>
<li class="nav-item"><a aria-expanded="true" class="dropdown-toggle nav-link" data-toggle="collapse" href="#volume4">Volume IV</a>
<ul class="collapse" id="volume4">
<li><a href="/examples/latest/Volumeiv.html">Contents</a></li>
<li><a href="/examples/latest/SeedsDataCloning.html" title="Estimating MLEs">Seeds</a></li>
<li><a href="/examples/latest/Coins.html" title="The ABC">Coins</a></li>
<li><a href="/examples/latest/SmartPhones.html" title="A bootstrap approach">Smart phones</a></li>
<li><a href="/examples/latest/Abbey.html" title="Share prices a stable approach">Abbey</a></li>
<!--<li><a href="/examples/latest/BeetlesMLE.html" title="Differential evolution and MLEs">Beetles</a></li>-->
<li><a href="/examples/latest/BeetlesProbit.html" title="Probit regression and censoring">Beetles</a></li>
<li><a href="/examples/latest/Preeclampsia.html" title="A multistage approach">Preeclampsia</a></li>
<li><a href="/examples/latest/Lotka-Volterra.html" title="A simple ODE model">Lotka-Volterra</a></li>
<li><a href="/examples/latest/FiveCompartment.html" title="A complex ODE model">Five compartment</a></li>
<li><a href="/examples/latest/ChangePoints.html" title="An ODE model with change points">Change points</a></li>
<li><a href="/examples/latest/Pollution.html" title="An autoregressive time series model">Pollution</a></li>
<li><a href="/examples/latest/Methadone.html" title="A large e-health random effects model">Methadone</a></li>
<li><a href="/examples/latest/Functionals.html" title="Integration and algebraic equations">Functionals</a></li>
</ul>
<li class="nav-item"><a aria-expanded="true" class="dropdown-toggle nav-link" data-toggle="collapse" href="#ecology">Ecology</a>
<ul class="collapse" id="ecology">
<li><a href="/examples/latest/VolumeEco.html">Contents</a></li>
<li><a href="/examples/latest/Gentians.html" title="Imperfect detection in site occppancy">Gentians</a></li>
<li><a href="/examples/latest/Sparrowhawk.html" title="Generalised metapopulation model">Sparrowhawks</a></li>
<li><a href="/examples/latest/Birds.html" title="Estimating species richness">Birds</a></li>
<li><a href="/examples/latest/Lizards.html" title="Mixture model for abundance with imperfect detection">Lizards</a></li>
<li><a href="/examples/latest/Voles.html" title="Estimating size of closed population">Voles</a></li>
<li><a href="/examples/latest/Impala.html" title="Distance sampling">Impala</a></li>
</ul>
</li>
<li class="nav-item"><a aria-expanded="true" class="dropdown-toggle nav-link" data-toggle="collapse" href="#geobugs">GeoBUGS</a>
<ul class="collapse" id="geobugs">
<li><a href="/examples/latest/spatial/Examples.html">Contents</a></li>
<li><a href="/examples/latest/spatial/Scotland.html" title="Lip cancer in Scotland">Conditional Autoregressive (CAR) models for disease mapping</a></li>
<li><a href="/examples/latest/spatial/LHA.html" title="Lung cancer in a London Health Authority">Convolution priors</a></li>
<li><a href="/examples/latest/spatial/Scotland1.html" title="Lip cancer revisited">Proper CAR model</a></li>
<li><a href="/examples/latest/spatial/Elevation.html" title="Surface elevation">Bayesian kriging and spatial prediction</a></li>
<li><a href="/examples/latest/spatial/Forest.html" title="Distribution of hickory trees in Duke forest">Poisson-gamma spatial moving average (convolution) model</a></li>
<li><a href="/examples/latest/spatial/Huddersfield.html" title="Ecological regression of air pollution and respiratory illness in children">Poisson-gamma spatial moving average (convolution) model</a></li>
<li><a href="/examples/latest/spatial/Shared.html" title="Oral cavity cancer and lung cancer in West Yorkshire (UK)">Shared component for mapping multiple diseases</a></li>
<li><a href="/examples/latest/spatial/Pollution.html">Random walk priors for temporal smoothing of daily air pollution estimates</a></li>
</ul>
</li>
<li class="nav-item"><a aria-expanded="true" class="dropdown-toggle nav-link" data-toggle="collapse" href="#reliability">ReliaBUGS</a>
<ul class="collapse" id="reliability">
<li><a href="/examples/latest/reliability/Vol.html">Contents</a></li>
<li><a href="/examples/latest/reliability/BS_Ex.html" title="'psi31' data from B&amp;S(1969)">Birnbuam-Saunders model</a></li>
<li><a href="/examples/latest/reliability/BurrX_Ex.html" title="'Ball bearing' data from Lawless(2003)">BurrX model</a></li>
<li><a href="/examples/latest/reliability/BurrXII_Ex.html" title="Simulated example ">BurrXII model</a></li>
<li><a href="/examples/latest/reliability/ExpoWeibull_Ex.html" title="Carbon Fibres data from Nichols and Padgett (2006) ">Two Parameter Exponentiated Weibull Example model</a></li>
<li><a href="/examples/latest/reliability/ExpPower_Ex.html" title="'Simulated data">Exponential Power model</a></li>
<li><a href="/examples/latest/reliability/Ext_exponential_Ex.html" title="'Carbon Fibres data' from Nichols and Padgett (2006)">Marshal-Olkin Extended Exponential (MOEE) model</a></li>
<li><a href="/examples/latest/reliability/Ext_Weibull_Ex.html" title="'Device failure times data' from Meeker and Escobar(1998)">Marshal-Olkin Extended Weibull model</a></li>
<li><a href="/examples/latest/reliability/FlexibleWeibull_Ex.html" title="'Reactor pump failure data' from Bebbinton, Lai, and Zitikis(2007)">Flexible Weibull model</a></li>
<li><a href="/examples/latest/reliability/GenExponential_Cen_Ex.html" title="'Ball bearing' data from Lawless(2003)">Censored Generalized Exponential model</a></li>
<li><a href="/examples/latest/reliability/Gompertz_Ex.html" title="Data from Murthy, Xie, and Jiang(2004)">Gompertz model</a></li>
<li><a href="/examples/latest/reliability/GPWeibull_Ex.html" title="'Carbon Fibres data' from Nichols and Padgett (2006) ">Two Parameter (GP) Exponentiated Weibull Example model</a></li>
<li><a href="/examples/latest/reliability/Gumbel_Ex.html" title="Data from Murthy, Xie, and Jiang(2004)">Gumbel model</a></li>
<li><a href="/examples/latest/reliability/Inv_Weibull_Ex.html" title="Data from Murthy, Xie, and Jiang(2004)">Inverse Weibull model</a></li>
<li><a href="/examples/latest/reliability/InvGaussian_Ex.html" title="'Transceiver data' from Chikara and Folks (9177)">Inverse Gaussian model</a></li>
<li><a href="/examples/latest/reliability/LinearFailure_Ex.html" title="Data from Murthy, Xie, and Jiang(2004)">Linear Failure Rate (LFR) model</a></li>
<li><a href="/examples/latest/reliability/LogisticExponential_Ex.html" title="'Naval logistics' data from Yan and Leemis (2008)">Logistic-Exponential model</a></li>
<li><a href="/examples/latest/reliability/Loglogistic_Ex.html" title="Simulated data">Log-Logistic model</a></li>
<li><a href="/examples/latest/reliability/LogWeibull_Ex.html" title="Data from Murthy, Xie, and Jiang(2004)">Log-Weibull model</a></li>
<li><a href="/examples/latest/reliability/ModifiedWeibull_Ex.html" title="Data from Aarset(1987)">Modified Weibull model</a></li>
<li><a href="/examples/latest/reliability/WeibullShifted_Ex.html" title="Simulated data">Weibull-shifted model</a></li>
</ul>
</li>
</li></li></li></li></ul>
</div>
</div></div><div class="col-12 col-md-9 col-lg-7 col-xl-6">


<p>
<h4>Pigs: genetic counseling and pedigree analysis</h4><br/>Spiegelhalter (1990) uses exact methods to analyse a small pedigree. This pedigree was previously used by Cannings and Thompson (1981) to illustrate their 'peeling' procedure to provide likelihoods for gene frequencies and probabilities for individuals being affected or carriers. We assume the pedigree refers to pigs which have the possibility of carrying a recessive gene: thus each pig has a genotype a<font color="#000100" face="Arial" size="2"><sub>1</sub></font>a<font color="#000100" face="Arial" size="2"><sub>1</sub></font>, a<font color="#000100" face="Arial" size="2"><sub>1</sub></font>a<font color="#000100" face="Arial" size="2"><sub>2</sub></font> or a<font color="#000100" face="Arial" size="2"><sub>2</sub></font>a<font color="#000100" face="Arial" size="2"><sub>2</sub></font>, in which only those with a<font color="#000100" face="Arial" size="2"><sub>2</sub></font>a<font color="#000100" face="Arial" size="2"><sub>2</sub></font> are affected with the trait, while those with a<font color="#000100" face="Arial" size="2"><sub>1</sub></font>a<font color="#000100" face="Arial" size="2"><sub>2</sub></font> are carriers of the defective allele a<font color="#000100" face="Arial" size="2"><sub>2</sub></font>. We assume that Ian (the consequence of a mating between Fred and his Aunt Clare) is yet to be born, and all that is known is that Fred's niece Jane has the trait. We wish to estimate the prevalence p of the allele a<font color="#000100" face="Arial" size="2"><sub>2</sub></font>, and predict the chance of Ian being affected. The conditional probability distributions are as follows. For the genotype of the founder nodes Ann, Brian, Eric and Henry we assume a binomial distribution<br/><br/>   Founder   ~   Binomial(q, 2)<br/><br/>where Founder takes values 0, 1 or 2 for genotypes a<font color="#000100" face="Arial" size="2"><sub>2</sub></font>a<font color="#000100" face="Arial" size="2"><sub>2</sub></font>, a<font color="#000100" face="Arial" size="2"><sub>1</sub></font>a<font color="#000100" face="Arial" size="2"><sub>2</sub></font> and a<font color="#000100" face="Arial" size="2"><sub>1</sub></font>a<font color="#000100" face="Arial" size="2"><sub>1</sub></font> respectively, and q = 1<font>- </font>p is the prevalence of the allele a<font color="#000100" face="Arial" size="2"><sub>1</sub></font>. This is equivalent to assuming Hardy Weinberg equilibrium, giving P(a<font color="#000100" face="Arial" size="2"><sub>1</sub></font>a<font color="#000100" face="Arial" size="2"><sub>1</sub></font>) = q<font color="#000100" face="Arial" size="2"><sup>2</sup></font>, P(a<font color="#000100" face="Arial" size="2"><sub>1</sub></font>a<font color="#000100" face="Arial" size="2"><sub>2</sub></font>) =2 q (1<font>- </font>q) , P(a<font color="#000100" face="Arial" size="2"><sub>2</sub></font> a<font color="#000100" face="Arial" size="2"><sub>2</sub></font>) = (1<font>- </font>q)<font color="#000100" face="Arial" size="2"><sup>2</sup></font>.<br/><br/>For the genotype of offspring we have the standard Mendelian inheritance probabilities given by the following table. <br/></p><center><br/><img alt="[pigs1]" src="pigs1.bmp"/><br/></center><br/>For a recessive gene the genotype-to-phenotype penetrance probabilities are given by:<br/><center><br/><br/><img alt="[pigs2]" src="pigs2.bmp"/><br/></center>The necessary inheritance probabilities are read in from the data file as an array<br/><br/>BUGS code for Pigs model:<br/>model <br/>{<br/>q ~ dunif(0,1) # prevalence of a1<br/>p &lt;- 1 - q # prevalence of a2<br/>Ann1 ~ dbin(q,2); Ann &lt;- Ann1 + 1 # geno. dist. for founder<br/>Brian1 ~ dbin(q,2); Brian &lt;- Brian1 + 1 <br/>Clare ~ dcat(p.mendelian[Ann,Brian,]) # geno. dist. for child<br/>Diane ~ dcat(p.mendelian[Ann,Brian,]) <br/>Eric1 ~ dbin(q,2)<br/>Eric &lt;- Eric1 + 1 <br/>Fred ~ dcat(p.mendelian[Diane,Eric,]) <br/>Gene ~ dcat(p.mendelian[Diane,Eric,]) <br/>Henry1 ~ dbin(q,2)<br/>Henry &lt;- Henry1 + 1 <br/>Ian ~ dcat(p.mendelian[Clare,Fred,]) <br/>Jane ~ dcat(p.mendelian[Gene,Henry,]) <br/>A1 ~ dcat(p.recessive[Ann,]) # phenotype distribution<br/>B1 ~ dcat(p.recessive[Brian,]) <br/>C1 ~ dcat(p.recessive[Clare,]) <br/>D1 ~ dcat(p.recessive[Diane,]) <br/>E1 ~ dcat(p.recessive[Eric,]) <br/>F1 ~ dcat(p.recessive[Fred,]) <br/>G1 ~ dcat(p.recessive[Gene,]) <br/>H1 ~ dcat(p.recessive[Henry,]) <br/>I1 ~ dcat(p.recessive[Ian,]) <br/>J1 ~ dcat(p.recessive[Jane,]) <br/>a &lt;- equals(Ann, 2) # event that Ann is carrier<br/>b &lt;- equals(Brian, 2) <br/>c &lt;- equals(Clare, 2) <br/>d &lt;- equals(Diane, 2) <br/>e &lt;- equals(Eric, 2) ;<br/>f &lt;- equals(Fred, 2) <br/>g &lt;- equals(Gene, 2) <br/>h &lt;- equals(Henry, 2) <br/>for (J in 1:3) {<br/>i[J] &lt;- equals(Ian, J) # i[1] = a1 a1<br/># i[2] = a1 a2<br/># i[3] = a2 a2 (i.e. Ian affected)<br/>}    }<br/><a href="/examples/latest/Pigsdata.html"></a><div><h5>Data</h5><code>
list(p.mendelian = structure(.Data=c(1.0, 0.0, 0.0,<br/>                     0.5, 0.5, 0.0,<br/>                     0.0, 1.0, 0.0, <br/>                     0.5, 0.5, 0.0,<br/>                     0.25, 0.5, 0.25,<br/>                     0.0, 0.5, 0.5,<br/>                     0.0, 1.0, 0.0,       <br/>                  0.0, 0.5, 0.5,<br/>                     0.0, 0.0, 1.0), .Dim=c(3,3,3)),<br/>      p.recessive = structure(.Data=c(1.0, 0.0, <br/>                        1.0, 0.0,<br/>                        0.0, 1.0), .Dim=c(3,2)),                              <br/>   A1 = 1, B1 = 1, C1 = 1, D1 = 1, E1 = 1, F1 = 1, G1 = 1, H1 = 1, J1 = 2)
</code></div><a href="/examples/latest/Pigsinits.html"></a><div><h5>Inits for chain 1</h5><code>
list(Ann1 = 2, Brian1 = 2, Clare = 2, Diane = 2, Eric1 = 2, Fred = 2, Gene = 2, Henry1 = 2, Ian = 2, Jane = 3)
</code></div>    <a href="/examples/latest/Pigsinits1.html"></a><div><h5>Inits for chain 2</h5><code>
list(<br/>Ann1 = 2.0,<br/>Brian1 = 1.0,<br/>Clare = 3.0,<br/>Diane = 2.0,<br/>Eric1 = 2.0,<br/>Fred = 3.0,<br/>Gene = 3.0,<br/>Henry1 = 2.0,<br/>I1 = 2.0,<br/>Ian = 3.0,<br/>Jane = 3.0,<br/>q = 0.7381)<br/>
</code></div><br/><font color="#0000FF" face="Arial" size="5">Results</font><br/>One update took 0.06ms<br/><br/>Update methods<br/><br/><font color="#000100" face="Arial" size="2">      Updater type   Size   Depth<br/>   Ann1   discrete slice updater   1   2<br/>   Brian1   discrete slice updater   1   2<br/>   Clare   discrete slice updater   1   3<br/>   Diane   discrete slice updater   1   3<br/>   Eric1   discrete slice updater   1   2<br/>   Fred   discrete slice updater   1   4<br/>   Gene   discrete slice updater   1   4<br/>   Henry1   discrete slice updater   1   2<br/>   I1   univariate forward updater   1   -6<br/>   Ian   univariate forward updater   1   -5<br/>   Jane   discrete slice updater   1   5<br/>   q   conjugate beta updater   1   1<br/></font><br/><center><img alt="[pigs3]" src="pigs3.bmp"/><br/><br/>DIC direct parents crashes<br/></center>We note a number of important tricks. First, each genotype is a 3-valued categorical variable with conditional probabilities either determined by the binomial (Hardy-Weinberg equilibrium) distribution (for founder nodes) or from the Mendelian inheritance probabilities which are stored as a 3-dimensional array <font color="#000100" face="Courier" size="3">p.mendelian</font>. In the latter case, the genotype of the parents picks which row of the matrix is used for the distribution. However, the rows of this matrix are indexed by values 1, 2 or 3, whilst the genotypes of the founder nodes take values 0, 1 or 2. Since BUGS does not allow subscripts to be functions of variables, we must first add 1 to the genotype of the parents (for example, <font color="#000100" face="Courier" size="3">Ann = Ann1 + 1</font>) and use these new variables as subscripts to the matrix <font color="#000100" face="Courier" size="3">p.mendelian</font>. The genotype-to-phenotype distribution is handled similarly in a matrix <font color="#000100" face="Courier" size="3">p.recessive</font>. Second, the <font color="#000100" face="Courier" size="3">equals</font> function <font color="#000100" face="Courier" size="3">equals(Ann, 2)</font> allows the calculation of P(Ann's genotype = 2) (i.e. a carrier), whilst <font color="#000100" face="Courier" size="3">equals(Ian, J)</font> calculates P(Ian's genotype = J), where J=3 implies that Ian is affected.<br/><br/>Check of externalization first process<br/><br/><font color="#000100" face="Arial" size="2"><strong>      mean   median   sd   MC_error   val2.5pc   val97.5pc   start   sample   ESS<br/></strong></font><font color="#000100" face="Arial" size="2">   i[1]   0.5717   1.0   0.4948   0.001695   0.0   1.0   5001   100000   85192<br/>   i[2]   0.3685   0.0   0.4824   0.00153   0.0   1.0   5001   100000   99367<br/>   i[3]   0.05981   0.0   0.2371   8.037E-4   0.0   1.0   5001   100000   87047<br/>   p   0.6258   0.636   0.1608   7.119E-4   0.2935   0.9013   5001   100000   51034<br/></font><br/>second process<br/><font color="#000100" face="Arial" size="2"><strong>      mean   median   sd   MC_error   val2.5pc   val97.5pc   start   sample   ESS<br/></strong></font><font color="#000100" face="Arial" size="2">   i[1]   0.5717   1.0   0.4948   0.001695   0.0   1.0   5001   100000   85192<br/>   i[2]   0.3685   0.0   0.4824   0.00153   0.0   1.0   5001   100000   99367<br/>   i[3]   0.05981   0.0   0.2371   8.037E-4   0.0   1.0   5001   100000   87047<br/>   p   0.6258   0.636   0.1608   7.119E-4   0.2935   0.9013   5001   100000   51034<br/><br/></font><font color="#000100" face="Arial" size="2"><strong>      mean   median   sd   MC_error   val2.5pc   val97.5pc   start   sample   ESS<br/></strong></font><font color="#000100" face="Arial" size="2">   i[1]   0.5729   1.0   0.4947   0.001292   0.0   1.0   5001   200000   146575<br/>   i[2]   0.367   0.0   0.482   0.001155   0.0   1.0   5001   200000   174058<br/>   i[3]   0.06014   0.0   0.2377   6.045E-4   0.0   1.0   5001   200000   154670<br/>   p   0.6257   0.6365   0.1607   4.811E-4   0.2932   0.9007   5001   200000   111615<br/></font>

<p></p>


</div><div class="d-none d-lg-block col-lg-3 col-xl-3" id="onthispage"><div class="sticky-top stickymenu"><h5>On this page</h5></div></div></div></div></body></html>